lib/ProductOpener/Display.pm

```diff
@@ -7742,27 +7742,26 @@ sub display_possible_improvement_description($$) {
 
 	if ((defined $product_ref->{improvements_data}) and (defined $product_ref->{improvements_data}{$tagid})) {
 
-		# Comparison of product nutrition facts to other products of the same category
+		my $template_data_ref_improvement = {};
+
+		$template_data_ref_improvement->{tagid} = $tagid;
+		$template_data_ref_improvement->{product_ref_improvements_data} = $product_ref->{improvements_data};
```

Comment:
we should keep the checks if ($tagid =~ /^en:nutrition-(very-)?high/) { and if ($tagid =~ /^en:better-nutri-score/) { here, otherwise we will get undefined values for the sprintf function calls like sprintf(lang("value_for_the_category"), display_taxonomy_tag($lc, "categories", $product_ref->{improvements_data}{$tagid}{category}));